recipes/newrelic/apm/php/debian.yml

```diff
@@ -296,30 +309,85 @@ install:
+              fi
+
+              #
+              # Get the PHP Binary directory associated with this particular NR INI file.
+              sed_slash_ini=$(echo "${ini}"  | sed 's/\//\\\//g')
+              php_bin=$(sed '1,/final pi_inidir_.*='"${sed_slash_ini}"'/d;/php -i output follows/,$d' $NR_INSTALL_LOG | sed -n 's/.*pi_bin=\(.*\/\)/\1/p')
+              if [ -z "${php_bin}" ]; then
+              #
+              # When there is only one PHP detected, the info format is different,
+              # we could detect from the log file, but since there is only one,
+              # just use `php`.
+              #
+                php_bin="php"
+              fi
+
+              #
+              # Modify the fresh copy of a brand new New Relic INI file.
+              #
+              ini_full_name="${ini}/newrelic.ini"
+              if [ ! -f "$ini_full_name" ]; then
+                continue
+              fi
+              sed -i "s/newrelic.appname = \"PHP Application\"/newrelic.appname = \"{{.NR_PHP_APPLICATION}}\"/" $ini_full_name
+              if [ "{{.NEW_RELIC_REGION}}" = "STAGING" ]; then
+                sed -i 's/;newrelic.daemon.collector_host = ""/newrelic.daemon.collector_host = "staging-collector.newrelic.com"/' $ini_full_name
+                sed -i 's/;newrelic.loglevel = "info"/newrelic.loglevel = "verbosedebug"/' $ini_full_name
+                sed -i 's/;newrelic.daemon.loglevel = "info"/newrelic.daemon.loglevel = "debug"/' $ini_full_name
+              fi
+
+              #
+              # Determine the target directory that the symlink is pointing to.
+              #
+              target_ini_dir="/etc/php5/conf.d/"
+              if [ -x /usr/sbin/phpquery ]; then
+                target_ini_dir="${ini}/../../mods-available/"
+              elif [ -x /usr/sbin/php5endmod ]; then
+                target_ini_dir="/etc/php5/mods-available/"
+              fi
+              if [ ! -d $target_ini_dir ]; then
+                echo -e "{{.RED}}Warning: Unable to move ${ini_full_name} to ${target_ini_dir}{{.GRAY}}"
+                continue
+              fi
+              #
+              # Move newrelic.ini created by tarball to the name that the installer will use.
+              # This ensures we always get the most up to date newrelic.ini file on the
+              # system.
+              #
+              mv ${ini_full_name} ${target_ini_dir}
+
+              #
+              # We rely on the package installer to enable the newrelic module.
+              #
+
+              #
+              # Three important pieces of information: the php binary location, the php ini
+              # file location, the actual (not symlink) New Relic ini file location.
+              # If we have web directories, copy the three key bits of info to the web file;
+              # otherwise copy to the cli file.  We will save the php directory, the target
+              # directory where the ini file is, and the php bin directory.
```

Comment:
The either / or nature of this check feels a bit off to me. According to [this comment](https://github.com/newrelic/newrelic-php-agent/blob/main/agent/newrelic-install.sh#L928-L940) in `newrelic-install.sh`, when there are separate PHPs for web and cli, they often (but not always) point to a separate scan directory, but end up sharing their config by storing them in a common directory and then symlinking to them.

The debian package follows this convention by detecting a common conf directory, putting the ini file there, and then calling `phpenmod` to create the symlink to each of the sapi directories present on the system.

Favoring the web version of PHP over the cli version when sending sample transactions makes sense. If that is all we are doing here, then a comment update is all that is needed. If, however, we really are only putting the ini file in the web or the cli conf directory, then I think we need to change the recipe to match the installer.

Reply:
The comment needs to be clarified.